modules/home/transformations/filter-plugins.php

```diff
@@ -21,19 +21,20 @@ private function get_installed_plugins(): array {
 		return array_keys( $plugins );
 	}
 
-	private function set_add_on_installation_status() {
-		$add_ons = $this->home_screen_data['add_ons']['repeater'];
+	private function get_add_ons_installation_status( array $add_ons ): array {
 		$index = 0;
 
 		foreach ( $add_ons as $add_on ) {
 			$is_plugin = array_key_exists( 'file_path', $add_on );
 			$is_installed_plugin = $is_plugin && in_array( $add_on['file_path'], $this->get_installed_plugins() );
```

Comment:
do we need to run get_installed_plugins() for every iteration ?

Reply:
@TzviRabinovitch The json is stored in the transient, and is only regenerated once an hour. Does that answer your question?